salt/client/__init__.py

```diff
@@ -1283,6 +1283,13 @@ def get_iter_returns(
                 if raw['data']['return'] == {}:
                     continue
 
+                # if the minion throws an exception containing the word "return"
+                # the master will try to handle the string as a dict in the next
+                # step. Check if we have a string, log the issue and continue.
+                if isinstance(raw['data']['return'], six.string_types):
+                    log.error("unexpected return from minion: %s", raw)
```

Comment:
I am not sure this should be a log.error...  maybe a log.info instead?  @rallytime do you have an opinion?

Reply:
I believe this is the correct log level.